crates/runtime/src/http/v1.rs

```diff
@@ -387,6 +388,53 @@ pub(crate) mod datasets {
             },
         }
     }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    #[serde(rename_all = "lowercase")]
+    pub(crate) struct DatasetRefreshResponse {
+        pub message: String,
+    }
+
+    pub(crate) async fn refresh(
+        Extension(app): Extension<Arc<RwLock<Option<App>>>>,
+        Extension(_df): Extension<Arc<RwLock<DataFusion>>>,
+        Path(dataset_name): Path<String>,
+    ) -> Response {
+        let app_lock = app.read().await;
+        let Some(readable_app) = &*app_lock else {
+            return (status::StatusCode::INTERNAL_SERVER_ERROR).into_response();
+        };
+
+        let Some(dataset) = readable_app
+            .datasets
+            .iter()
+            .find(|d| d.name == dataset_name)
+        else {
+            return (
+                status::StatusCode::BAD_REQUEST,
```

Comment:
Probably 404 is more appropriate here.

Reply:
Yeah, good point @digadeesh - thank you. Double checked and GitHub API also returns a 404 when a specific repository, user, or other resource specified in the URL path does not exist.

Reply:
Updated 👍